src/oge/helpers.py

```diff
@@ -579,6 +614,136 @@ def add_plant_entity(df: pd.DataFrame) -> pd.DataFrame:
     return df
 
 
+def add_missing_location(df: pd.DataFrame) -> pd.DataFrame:
+    """Add missing latitude, longitude, state, county and city when possible.
+
+    Args:
+        df (pd.DataFrame): table with 'latitude', 'longitude', 'state', 'county' and
+            'city' columns
+
+    Returns:
+        pd.DataFrame: original data frame with missing 'latitude', 'longitude',
+            'state', 'county' and 'city' filled out when possible.
+    """
+    # get lat/lon
+    missing_coord = df[df["longitude"].isna() | df["latitude"].isna()]
+    if len(missing_coord) > 0:
+        # only get coordinates when state, county and city are available
+        for i in missing_coord.index:
+            state = df.loc[i, "state"]
+            county = df.loc[i, "county"]
+            city = df.loc[i, "city"]
+            # if county == "NOT IN FILE" or city == "unsited":
```

Comment:
Should this commented out code be removed?

Reply:
Yes

Reply:
Done